sherpa/astro/data.py

```diff
@@ -1439,6 +1486,13 @@ def __init__(self, name, channel, counts, staterror=None, syserror=None,
         self._plot_fac = 0
         self.units = 'channel'
         self.quality_filter = None
+        # Track the requested noticed/ignored data range, so that it
+        # can be re-created when swaping between grouped and ungrouped
+        # data.
+        #
+        # We start off with an explicit "notice everything" call, as this
+        # means less special cases. At least, that's the hope.
```

Comment:
I'm always encourage by these slight upbeat comments!

Reply:
The approach has so far seemed to work out well, so I'm just waiting for the other shoe to drop...

Or when I realize it won't actually help us in our plans to fix all the things™.